addons/payment_ogone/models/payment_token.py

```diff
@@ -0,0 +1,19 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+
+
+class PaymentToken(models.Model):
+    _inherit = 'payment.token'
+
+    def write(self, values):
+        """
+        Override to prevent archiving of tokens during Ogone to Worldline migration.
+
+        :param dict values: The values to write on the record.
+        :return: None if the write is skipped, otherwise the result of the parent write.
+        :rtype: bool or None
+        """
+        if self._context.get('skip_token_archival') and 'active' in values:
+            return
+        return super().write(values)
```

Comment:
I would override `_archive_linked_tokens` method instead this one in new module and filter ogone related token based on context 

P.S. use `self.env.context` instead `self._context` as it's quite old and deprecated in newer versions